tests/nvmeof/workflows/ha.py

```diff
@@ -942,6 +943,157 @@ def validate_initiator(self, gateway, ana_id, failed_gw=None):
             initiators for gateway {gateway.daemon_name}"
         )
 
+    def validate_init_namespace_masking(
+        self,
+        command,
+        init_nodes,
+        expected_visibility,
+        validate_config=None,
+    ):
+        """Validate that the namespace visibility is correct from all initiators."""
+        for node in init_nodes:
+            initiator_node = get_node_by_id(self.cluster, node)
+            client = NVMeInitiator(initiator_node, self.gateways[0])
+            client.disconnect_all()  # Reconnect NVMe targets
+            client.connect_targets(config={"nqn": "connect-all"})
+            devices_output, _ = initiator_node.exec_command(
+                cmd="nvme list --output-format=json", sudo=True
+            )  # Get NVMe device list in JSON format
+            devices_json = json.loads(devices_output)
+
+            def subsystem_nsid_found(dictionary, key, value):
+                return key in dictionary and value in dictionary[key]
+
+            if command == "add_host":
+                args = validate_config["args"]
+                subsystem_to_nsid = {args.get("nsid"): args.get("sub_num")}
+                init_node = args.get("init_node")
+                serial_to_namespace = defaultdict(list)
+
+                devices = devices_json.get("Devices", [])
+                if not devices:
+                    LOG.info(f"No devices found on node {node}")
+                    continue
+
+                for device in devices:
+                    key = device["NameSpace"]
+                    value = int(device["SerialNumber"])
+                    serial_to_namespace[key].append(value)
+
+                ns_to_check, subsystem_to_check = next(iter(subsystem_to_nsid.items()))
+                LOG.info(f"{subsystem_to_nsid} : {serial_to_namespace}")
+
+                if node == init_node:
+                    if subsystem_nsid_found(
```

Comment:
Do you think we need a retry here?

Reply:
I do no think it will make difference, but please explain how it helps?